### PR TITLE
Compatibility connector tests with native types and PSL features

### PR DESCRIPTION
**Pull request by carmenberndt**
closes https://github.com/prisma/prisma-engines/issues/1246
related: https://www.notion.so/prismaio/Compatibilities-with-Prisma-attributes-6e0902265bc94b8f88ccb47d780fcccb

### CODE REVIEW

**Review by dpetrick — APPROVED** [review]
Note: This is somewhat overlapping with the other native type tests that I wrote in the early stages of native types, these here might replace those completely because they're more extensive.

Otherwise my only real gripe is the amount of code duplication, but this is usually the case with the scala tests, unfortuntately. 

Anyways, looks good for now.

File: query-engine/connector-test-kit/src/test/scala/queries/nativeTypes/nativeTypesWithPSLFeaturesOnPostgres.scala [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+             |model Sun {
+             |  id Int @id
+             |  name     $p_type @test.$n_type
+             |}
+    """.stripMargin
+        }
+        assert(database.setupWithStatusCode(project) == 0)
+      }
+  }
+
```
Comment (do4gr) [inline query-engine/connector-test-kit/src/test/scala/queries/nativeTypes/nativeTypesWithPSLFeaturesOnPostgres.scala:358]: This is a Mysql error in the postgres file, maybe a copy artifact?

Comment (carmenberndt) [reply query-engine/connector-test-kit/src/test/scala/queries/nativeTypes/nativeTypesWithPSLFeaturesOnPostgres.scala:358]: Ah, yeah I copied the error message, but ` internal error: entered unreachable code: NOW default on non-datetime column"` is the same for both MySQL and Postgres

**Comment by tomhoule** [comment]
It's not clear why this was never merged (nobody remembers), but we have moved the whole connector-test-kit from scala to rust, and all tests have been rewritten. This is going to be hard to port. If @dpetrick is ok, let's close this PR.

**Pull request closed without merging**